chore: Skip files with empty controller classes

gavindsouza opened the pull request:
## Problem Statement

There are many empty controller files that are included in the coverage results at https://coveralls.io/github/frappe/frappe. Since those controllers are empty, they will default to the `Document` class which is already being covered/tested. 

## Changes

Include any file in the coverage report if it matches either of following conditions

- The controller class has any methods defined/overridden
- A global variable is defined in the module
- More than a single class is defined in the module
- At least one function is defined in the module

> This PR is an attempt to more accurately calculate the test coverage.

gavindsouza committed:
chore: Skip files with empty controller classes

This may not get the most accurate results since we're assuming the empty controller class to have < 10 LOC, but it's going to be "close"

ankush commented on frappe/parallel_test_runner.py:
```diff
@@ -134,8 +136,25 @@ def start_coverage(self):
 			]
 
 			if self.app == 'frappe':
```
can we do this for ERPNext too? At least the part about excluding `test_*` files? 🤔

gavindsouza replied:
If it helps to accurately calculate coverage 👍🏼

ankush replied:
Move ahead with merging? No one has stated any objection to this change.

gavindsouza committed:
refactor(coverage): Calculate files to be excluded

gavindsouza committed:
fix: Use frappe.coverage.get_exclude_list

Commonify interfaces for parrallel and standard test suite

gavindsouza committed:
style: Reverse PEP 263

gavindsouza committed:
Merge branch 'develop' of github.com:frappe/frappe into cov-skip-empty-files

gavindsouza committed:
refactor(minor): Excluding files via frappe.coverage

gavindsouza commented:
Okay, so this was pointless 😆

gavindsouza committed:
test: Delete empty controller files

to test hypothesis even though coveralls % stayed the same after the
last commit :')
